File: android/src/main/kotlin/net/allinmeta/flutter_core_channel/BaseMessengerHandler.kt
package net.allinmeta.flutter_core_channel

typealias FlutterMessengerMap = HashMap<String, Any>
typealias MessageResult = (result: FlutterMessengerMap?) -> Unit

/**
 * Flutter 消息处理器基类。外部继承实现此接口，即可处理指定通道方法的消息
 */
interface BaseMessengerHandler {

    /// 当前通道的调用的方法名称
    fun methodName(): String

    /// 收到来自Flutter的消息
    /// params: 消息参数
    /// action: 执行目的
    /// 返回给Flutter的值
    fun didReceivedFlutterSignal(params: FlutterMessengerMap?, action: String): FlutterMessengerMap?
}

File: ios/Classes/BaseMessengerHandler.swift
import Foundation
import Flutter
public typealias FlutterMessengerMap = [String: Any]
public typealias  MessageResult = ((FlutterMessengerMap?) -> Void)
@objc public protocol BaseMessengerHandler: NSObjectProtocol {
    func methodName() -> String
    func didReceivedFlutterSignal(params: FlutterMessengerMap?) -> FlutterMessengerMap?
}